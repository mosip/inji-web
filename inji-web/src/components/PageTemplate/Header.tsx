import React, {useState} from "react";
import {useTranslation} from "react-i18next";
import {useNavigate} from "react-router-dom";
import {LanguageSelector} from "../Common/LanguageSelector";
import {GiHamburgerMenu} from "react-icons/gi";
import OutsideClickHandler from 'react-outside-click-handler';

export const Header: React.FC = () => {

    const {t} = useTranslation("PageTemplate");
    const [isOpen, setIsOpen] = useState(false);
    const navigate = useNavigate();

    return (
        <header>
        <div data-testid="Header-Container"
                className="fixed top-0 left-0 right-0 bg-iw-background py-4 shadow-sm">
            <div className="container mx-auto flex justify-between items-center px-4">
                <div data-testid="Header-InjiWeb-Logo-Container" className={"flex flex-row justify-center items-center"}>
                    <div className={"sm:hidden"} onClick={() => setIsOpen(open => !open)}>
                        <GiHamburgerMenu size={32}/>
                    </div>
                    <img src={require("../../assets/InjiWebLogo.png")}
                         onClick={() => navigate("/")}
                         className={"h-13 w-28 cursor-pointer"}
                         data-testid="Header-InjiWeb-Logo"
                         alt="Inji Web Logo"/>
                </div>
                <nav>
                    <ul className="flex space-x-4 items-center font-semibold" data-testid="Header-Menu-Elements">
                        <li data-testid="Header-Menu-AboutInji"><a href="https://docs.mosip.io/inji/inji-web/overview"
                                                                   target="_blank"
                                                                   rel="noreferrer"
                                                                   className="text-iw-title hidden sm:inline-block">{t("Header.aboutInji")}</a>
                        </li>
                        <li data-testid="Header-Menu-Help">
                            <div data-testid="Header-Menu-Help-div" onClick={() => navigate("/help")}
                                 className="text-iw-title cursor-pointer hidden sm:inline-block">{t("Header.help")}</div>
                        </li>
                        <li data-testid="Header-Menu-LanguageSelector"><LanguageSelector/></li>
                    </ul>
                </nav>
            </div>
            { isOpen &&
                <OutsideClickHandler onOutsideClick={()=>setIsOpen(false)}>
                    <div className="container mx-auto px-4 flex flex-col justify-start items-start font-semibold" onBlur={() => {console.log("Blur is pressed");setIsOpen(false)}}>
                        <div data-testid="Header-Menu-AboutInji" className={"py-5 w-full"}
                             onClick={() => {window.open("https://docs.mosip.io/inji/inji-web/overview");setIsOpen(false)}}>
                            {t("Header.aboutInji")}
                        </div>
                        <div data-testid="Header-Menu-Help"
                             onClick={() => {navigate("/help");setIsOpen(false)}}
                             className="text-iw-title cursor-pointer py-5 w-full inline-block sm:hidden">
                                {t("Header.help")}
                        </div>
                    </div>
                </OutsideClickHandler>
            }
        </div>
        </header>
    )

}

