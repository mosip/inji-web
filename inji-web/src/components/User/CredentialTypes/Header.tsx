import React from "react";
import {SearchCredential} from "../../Credentials/SearchCredential";
import {NavBackArrowButton} from "../../Common/Buttons/NavBackArrowButton";
import {IssuerWellknownDisplayArrayObject} from "../../../types/data";
import {useTranslation} from "react-i18next";
import {TertiaryButton} from "../../Common/Buttons/TertiaryButton";
import {HeaderStyles} from "./HeaderStyles";

interface HeaderPops {
    onBackClick: () => void,
    displayObject: IssuerWellknownDisplayArrayObject | undefined,
    onClick: () => void,
}

export const Header: React.FC<HeaderPops> = (props) => {
    const {t} = useTranslation('User');
    return <div className={HeaderStyles.headerContainer}>
        <div className={HeaderStyles.headerLeftSection}>
            <div className={HeaderStyles.headerLeftSection}>
                <NavBackArrowButton onBackClick={props.onBackClick}/>
            </div>
            <div className={HeaderStyles.headerTitleSection}>
                <span
                    data-testid={"Stored-Credentials"}
                    className={HeaderStyles.pageTitle}
                >
                    {props.displayObject?.name}
                </span>

                <TertiaryButton onClick={props.onClick} title={t('Common:home')} testId={"home"}/>
            </div>
        </div>
        <div>
            <SearchCredential
                issuerContainerBorderRadius={"rounded-md"}
            />
        </div>
    </div>;
}