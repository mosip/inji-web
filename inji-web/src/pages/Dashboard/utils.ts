export const convertStringIntoPascalCase = (text: string | null) => {
    if (text != null) {
        return text
            .toLocaleLowerCase()
            .split(" ")
            .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
            .join(" ");
    }
};
