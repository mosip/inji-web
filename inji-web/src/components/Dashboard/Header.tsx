import React, {useState, useEffect, useRef} from 'react';
import {useNavigate} from 'react-router-dom';
import {LanguageSelector} from '../Common/LanguageSelector';
import {api} from '../../utils/api';
import {useCookies} from 'react-cookie';
import {toast} from 'react-toastify';
import {
    convertStringIntoPascalCase,
    navigateToDashboardHome
} from '../../pages/Dashboard/utils';
import {useUser} from '../../hooks/useUser';
import {DashboardHeaderProps, DropdownItem} from './types';
import HamburgerMenu from '../../assets/HamburgerMenu.svg';
import {isRTL} from '../../utils/i18n';
import {useSelector} from 'react-redux';
import {RootState} from '../../types/redux';
import {useTranslation} from 'react-i18next';
import {getProfileInitials} from './Utils';
import DropdownArrowIcon from '../Common/DropdownArrowIcon';
import {KEYS} from '../../utils/constants';

export const Header: React.FC<DashboardHeaderProps> = ({
    headerRef,
    headerHeight
}) => {
    const language = useSelector((state: RootState) => state.common.language);
    const navigate = useNavigate();
    const [displayName, setDisplayName] = useState<string | undefined>(
        undefined
    );
    const [isProfileDropdownOpen, setIsProfileDropdownOpen] = useState(false);
    const [isHamburgerMenuOpen, setIsHamburgerMenuOpen] = useState(false);
    const [cookies] = useCookies(['XSRF-TOKEN']);
    const dropdownRef = useRef<HTMLDivElement>(null);
    const hamburgerMenuRef = useRef<HTMLImageElement>(null);
    const {user, removeUser} = useUser();
    const displayNameFromLocalStorage = user?.displayName;
    const hasProfilePictureUrl = user?.profilePictureUrl;
    const {t} = useTranslation('Dashboard');

    useEffect(() => {
        setDisplayName(displayNameFromLocalStorage);
    }, [displayNameFromLocalStorage]);

    useEffect(() => {
        const handleClickOutside = (event: MouseEvent) => {
            const target = event.target as Node;

            if (dropdownRef.current && !dropdownRef.current.contains(target)) {
                setIsProfileDropdownOpen(false);
            }

            if (
                hamburgerMenuRef.current &&
                !hamburgerMenuRef.current.contains(target)
            ) {
                setIsHamburgerMenuOpen(false);
            }
        };

        document.addEventListener('click', handleClickOutside);
        return () =>
            document.removeEventListener('click', handleClickOutside);
    }, []);

    const handleLogout = async () => {
        try {
            const apiRequest = api.userLogout;
            const response = await fetch(apiRequest.url(), {
                method: 'POST',
                credentials: 'include',
                headers: {
                    'X-XSRF-TOKEN': cookies['XSRF-TOKEN']
                }
            });

            if (response.ok) {
                removeUser();
                localStorage.removeItem(KEYS.WALLET_ID);
                window.location.replace('/');
            } else {
                const parsedResponse = await response.json();
                const errorCode = parsedResponse?.errors[0].errorCode;
                if (errorCode === 'user_logout_error') {
                    removeUser();
                    window.location.replace('/');
                }
                throw new Error(parsedResponse?.errors[0]?.errorMessage);
            }
        } catch (error) {
            console.error('Logout failed with error:', error);
            toast.error('Unable to logout. Please try again.');
        }
    };

    const dropdownItems: DropdownItem[] = [
        {
            label: t('ProfileDropdown.profile'),
            onClick: () => {
                setIsProfileDropdownOpen(false);    
                navigate('profile',{state: {from: window.location.pathname}});
            },
            textColor: 'text-gray-700',
            key: 'Profile-Dropdown-Profile'
        },
        {
            label: t('ProfileDropdown.faq'),
            onClick: () => {
                setIsProfileDropdownOpen(false);
                navigate('faq', {state: {from: window.location.pathname}});
            },
            textColor: 'text-gray-700',
            key: 'Profile-Dropdown-FAQ'
        },
        {
            label: t('ProfileDropdown.logout'),
            onClick: handleLogout,
            textColor: 'text-red-700',
            key: 'Profile-Dropdown-Logout'
        }
    ];

    const toggleProfileDropdown = () => {
        setIsProfileDropdownOpen(!isProfileDropdownOpen);
    };

    const toggleHamburgerMenu = () => {
        setIsHamburgerMenuOpen(!isHamburgerMenuOpen);
    };

    const getUserProfileIconWithName = () => (
        <div className="flex gap-2 items-center">
            <div
                className={`aspect-square w-12 sm:w-14 rounded-full bg-[#DDDDDD] overflow-hidden flex items-center justify-center text-[#1A001D] font-medium text-lg ${
                    !hasProfilePictureUrl && 'p-2 sm:p-3 md:p-4'
                }`}
            >
                {hasProfilePictureUrl ? (
                    <img
                        src={user.profilePictureUrl}
                        alt="Profile Pic"
                        className="w-full h-full object-cover rounded-full"
                        referrerPolicy="no-referrer"
                    />
                ) : (
                    getProfileInitials(displayName)
                )}
            </div>
            <span className="font-semibold text-gray-800">
                {convertStringIntoPascalCase(displayName)}
            </span>
        </div>
    );

    return (
        <header
            ref={headerRef}
            data-testid="Dashboard-Header-Container"
            className="fixed top-0 left-0 right-0 z-20 bg-iw-background bg-transparent shadow-[0_4px_5px_0_rgba(0,0,0,0.051)]"
        >
            <div className="container mx-auto px-4 py-4 flex justify-between items-center">
                <div className="flex justify-start items-center gap-2">
                    <div
                        data-testid="Hamburger-Menu"
                        className="block sm:hidden"
                    >
                        <img
                            data-testid="Hamburger-Menu-icon"
                            ref={hamburgerMenuRef}
                            src={HamburgerMenu}
                            alt="Hamburger Menu"
                            onClick={toggleHamburgerMenu}
                            className="cursor-pointer"
                        />
                    </div>
                    <div
                        className="w-[130px] sm:w-[160px] md:w-[170px]"
                        role={'button'}
                        tabIndex={0}
                        onMouseDown={() => navigateToDashboardHome(navigate)}
                        onKeyUp={() => navigateToDashboardHome(navigate)}
                    >
                        <img
                            src={require('../../assets/InjiWebLogo.png')}
                            className={`max-w-full h-auto object-contain cursor-pointer ${
                                isRTL(language) ? 'mr-4' : ''
                            }`}
                            data-testid="Header-InjiWeb-Logo"
                            alt="Inji Web Logo"
                        />
                    </div>
                </div>

                <div className="flex items-center space-x-6">
                    <LanguageSelector />

                    <div
                        data-testid="Profile-Details"
                        className="hidden sm:block relative"
                        ref={dropdownRef}
                    >
                        <div className="flex items-center space-x-2">
                            {getUserProfileIconWithName()}
                            <div
                                className="relative inline-block cursor-pointer"
                                onClick={(e) => {
                                    e.stopPropagation(); // Prevent global click handler from immediately closing the dropdown
                                    toggleProfileDropdown();
                                  }}
                            >
                                <DropdownArrowIcon
                                    isOpen={isProfileDropdownOpen}
                                />
                            </div>
                        </div>

                        {isProfileDropdownOpen && (
                            <div
                                data-testid="Profile-Dropdown"
                                className="absolute -right-7 top-100 mt-8 w-56 bg-white rounded-lg shadow-lg z-50 font-medium"
                            >
                                <div className="absolute top-[-0.3rem] right-8 w-3 h-3 bg-white transform rotate-45" />
                                <div className="py-2">
                                    {dropdownItems.map((item, index) => (
                                        <React.Fragment key={item.key}>
                                            <div
                                                className={`px-4 py-2 text-sm ${item.textColor} hover:bg-gray-100 cursor-pointer`}
                                                onClick={item.onClick}
                                            >
                                                {item.label}
                                            </div>
                                            {index !==
                                                dropdownItems.length - 1 && (
                                                <hr className="border-gray-200 my-1 mx-2" />
                                            )}
                                        </React.Fragment>
                                    ))}
                                </div>
                            </div>
                        )}
                    </div>
                </div>
            </div>
            <div
                data-testid="Hamburger-Menu-dropdown"
                className="block sm:hidden w-full"
            >
                {isHamburgerMenuOpen && (
                    <div
                        style={{marginTop: headerHeight}}
                        className="absolute top-1 bg-white shadow-iw-hamburger-dropdown p-2 w-full"
                    >
                        <div>
                            <div className="flex items-center px-4 py-2">
                                {getUserProfileIconWithName()}
                            </div>
                            {dropdownItems.map((item, index) => (
                                <React.Fragment key={item.key}>
                                    <div
                                        className={`px-4 py-2 text-sm ${item.textColor} hover:bg-gray-100 cursor-pointer font-medium`}
                                        onClick={item.onClick}
                                    >
                                        {item.label}
                                    </div>
                                    {index !== dropdownItems.length - 1 && (
                                        <hr className="border-gray-200 my-1 mx-2" />
                                    )}
                                </React.Fragment>
                            ))}
                        </div>
                    </div>
                )}
            </div>
        </header>
    );
};
