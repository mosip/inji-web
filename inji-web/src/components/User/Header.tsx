import React, {useEffect, useRef, useState} from 'react';
import {useNavigate} from 'react-router-dom';
import {LanguageSelector} from '../Common/LanguageSelector';
import {api} from '../../utils/api';
import {toast} from 'react-toastify';
import {useUser} from '../../hooks/User/useUser';
import HamburgerMenu from '../../assets/HamburgerMenu.svg';
import {isRTL} from '../../utils/i18n';
import {useSelector} from 'react-redux';
import {RootState} from '../../types/redux';
import {useTranslation} from 'react-i18next';
import DropdownArrowIcon from '../Common/DropdownArrowIcon';
import {ROUTES} from '../../utils/constants';
import {convertStringIntoPascalCase} from "../../utils/misc";
import {navigateToUserHome} from "../../utils/navigationUtils";
import {CircleSkeleton} from '../Common/CircleSkeleton';
import {InfoFieldSkeleton} from '../Common/InfoFieldSkeleton';
import {ApiError, DropdownItem, ResponseTypeObject} from "../../types/data";
import {useApi} from "../../hooks/useApi";

type UserHeaderProps = {
    headerRef: React.RefObject<HTMLDivElement>;
    headerHeight: number;
};

export const Header: React.FC<UserHeaderProps> = ({
    headerRef,
    headerHeight
}) => {
    const language = useSelector((state: RootState) => state.common.language);
    const navigate = useNavigate();
    const [displayName, setDisplayName] = useState<string | undefined>(
        undefined
    );
    const [isProfileDropdownOpen, setIsProfileDropdownOpen] = useState(false);
    const [isHamburgerMenuOpen, setIsHamburgerMenuOpen] = useState(false);
    const dropdownRef = useRef<HTMLDivElement>(null);
    const hamburgerMenuRef = useRef<HTMLImageElement>(null);
    const {user, removeUser,isLoading} = useUser();
    const displayNameFromLocalStorage = user?.displayName;
    const hasProfilePictureUrl = user?.profilePictureUrl;
    const {t} = useTranslation('User');
    const logoutRequest = useApi()

    useEffect(() => {
        setDisplayName(displayNameFromLocalStorage);
    }, [displayNameFromLocalStorage]);

    useEffect(() => {
        const handleClickOutside = (event: MouseEvent) => {
            const target = event.target as Node;

            if (dropdownRef.current && !dropdownRef.current.contains(target)) {
                setIsProfileDropdownOpen(false);
            }

            if (
                hamburgerMenuRef.current &&
                !hamburgerMenuRef.current.contains(target)
            ) {
                setIsHamburgerMenuOpen(false);
            }
        };

        document.addEventListener('click', handleClickOutside);
        return () =>
            document.removeEventListener('click', handleClickOutside);
    }, []);

    const handleLogout = async () => {
        try {
            const response = await logoutRequest.fetchData({
                apiRequest: api.userLogout,
            })

            if (response.ok()) {
                removeUser();
                window.location.replace(ROUTES.ROOT);
            } else {
                const parsedResponse = ((response.error as ApiError)?.response?.data as ResponseTypeObject).errors
                const errorCode = parsedResponse?.[0].errorCode;
                if (errorCode === 'user_logout_error') {
                    removeUser();
                    window.location.replace(ROUTES.ROOT);
                }
                throw new Error(parsedResponse?.[0]?.errorMessage);
            }
        } catch (error) {
            console.error('Logout failed with error:', error);
            toast.error('Unable to logout. Please try again.');
        }
    };

    const dropdownItems: DropdownItem[] = [
        {
            label: t('ProfileDropdown.profile'),
            onClick: () => {
                setIsProfileDropdownOpen(false);    
                navigate(ROUTES.PROFILE,{state: {from: window.location.pathname}});
            },
            textColor: 'text-gray-700',
            key: 'Profile-Dropdown-Profile'
        },
        {
            label: t('ProfileDropdown.faq'),
            onClick: () => {
                setIsProfileDropdownOpen(false);
                navigate(ROUTES.USER_FAQ, {state: {from: window.location.pathname}});
            },
            textColor: 'text-gray-700',
            key: 'Profile-Dropdown-FAQ'
        },
        {
            label: t('ProfileDropdown.logout'),
            onClick: handleLogout,
            textColor: 'text-red-700',
            key: 'Profile-Dropdown-Logout'
        }
    ];

    const toggleProfileDropdown = (e: React.MouseEvent<HTMLDivElement>) => {
        e.stopPropagation(); // Prevent global click handler from immediately closing the dropdown
        setIsProfileDropdownOpen(!isProfileDropdownOpen);
    };

    const toggleHamburgerMenu = () => {
        setIsHamburgerMenuOpen(!isHamburgerMenuOpen);
    };

    const getProfileInitials = (displayName: string | undefined) => {
        return displayName ? displayName.charAt(0).toUpperCase(): 'U';
    };

    const getUserProfileIconWithName = () => {
        if (isLoading) {
            return (
              <div className="flex gap-2 items-center">
                <CircleSkeleton size="w-12 h-12" />
                <InfoFieldSkeleton width="w-24" height="h-2" />
              </div>
            );
          }
        
          return (
            <div className="flex gap-2 items-center">
              <div
                className={`aspect-square w-12 rounded-full bg-iw-avatarPlaceholder overflow-hidden flex items-center justify-center text-iw-avatarText font-medium text-lg ${
                  !hasProfilePictureUrl && 'p-2 sm:p-3 md:p-4'
                }`}
              >
                {hasProfilePictureUrl ? (
                  <img
                    src={user.profilePictureUrl}
                    alt="Profile Pic"
                    className="w-12 h-12 object-cover rounded-full"
                    referrerPolicy="no-referrer"
                  />
                ) : (
                  getProfileInitials(displayName)
                )}
              </div>
              <span className="font-semibold text-gray-800">
                {convertStringIntoPascalCase(displayName)}
              </span>
            </div>
          );
        };

    return (
        <header
            ref={headerRef}
            data-testid="dashboard-header-container"
            className="fixed top-0 left-0 right-0 z-20 bg-iw-background bg-transparent shadow-[0_4px_5px_0_rgba(0,0,0,0.051)]"
        >
            <div className="container mx-auto px-4 py-4 flex justify-between items-center">
                <div className="flex justify-start items-center gap-2">
                    <div
                        data-testid="hamburger-menu"
                        className="block sm:hidden"
                    >
                        <img
                            data-testid="icon-hamburger-menu"
                            ref={hamburgerMenuRef}
                            src={HamburgerMenu}
                            alt="Hamburger Menu"
                            onClick={toggleHamburgerMenu}
                            className="cursor-pointer"
                        />
                    </div>
                    <div
                        className="w-[130px] sm:w-[160px] md:w-[170px]"
                        role={'button'}
                        tabIndex={0}
                        onMouseDown={() => navigateToUserHome(navigate)}
                        onKeyUp={() => navigateToUserHome(navigate)}
                    >
                        <img
                            src={require('../../assets/InjiWebLogo.png')}
                            className={`max-w-full h-auto object-contain cursor-pointer ${
                                isRTL(language) ? 'mr-4' : ''
                            }`}
                            data-testid="header-injiWeb-logo"
                            alt="Inji Web Logo"
                        />
                    </div>
                </div>

                <div className="flex items-center space-x-6">
                    <LanguageSelector />

                    <div
                        data-testid="profile-details"
                        className="hidden sm:block relative"
                        ref={dropdownRef}
                    >
                        <div className="flex items-center space-x-2">
                            {getUserProfileIconWithName()}
                            {!isLoading && (
                                <div
                                className="relative inline-block cursor-pointer"
                                onClick={toggleProfileDropdown}
                                >
                                <DropdownArrowIcon isOpen={isProfileDropdownOpen} />
                                </div>
                            )}
                        </div>

                        {isProfileDropdownOpen && (
                            <div
                                data-testid="profile-dropdown"
                                className="absolute -right-7 top-100 mt-8 w-56 bg-white rounded-lg shadow-lg z-50 font-medium"
                            >
                                <div className="absolute top-[-0.3rem] right-8 w-3 h-3 bg-white transform rotate-45" />
                                <div className="py-2">
                                    {dropdownItems.map((item, index) => (
                                        <React.Fragment key={item.key}>
                                            <div
                                                className={`px-4 py-2 text-sm ${item.textColor} hover:bg-gray-100 cursor-pointer`}
                                                onClick={item.onClick}
                                            >
                                                {item.label}
                                            </div>
                                            {index !==
                                                dropdownItems.length - 1 && (
                                                <hr className="border-gray-200 my-1 mx-2" />
                                            )}
                                        </React.Fragment>
                                    ))}
                                </div>
                            </div>
                        )}
                    </div>
                </div>
            </div>
            <div
                data-testid="hamburger-menu-dropdown"
                className="block sm:hidden w-full"
            >
                {isHamburgerMenuOpen && (
                    <div
                        style={{marginTop: headerHeight}}
                        className="absolute top-1 bg-white shadow-iw-hamburger-dropdown p-2 w-full"
                    >
                        <div>
                            <div className="flex items-center px-4 py-2">
                                {getUserProfileIconWithName()}
                            </div>
                            {dropdownItems.map((item, index) => (
                                <React.Fragment key={item.key}>
                                    <div
                                        className={`px-4 py-2 text-sm ${item.textColor} hover:bg-gray-100 cursor-pointer font-medium`}
                                        onClick={item.onClick}
                                    >
                                        {item.label}
                                    </div>
                                    {index !== dropdownItems.length - 1 && (
                                        <hr className="border-gray-200 my-1 mx-2" />
                                    )}
                                </React.Fragment>
                            ))}
                        </div>
                    </div>
                )}
            </div>
        </header>
    );
};
