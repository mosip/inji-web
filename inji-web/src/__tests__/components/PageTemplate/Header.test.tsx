import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react';
import {Header} from "../../../components/PageTemplate/Header";
import {mockUseNavigate, wrapUnderRouter} from "../../../utils/mockUtils";

const mockedUsedNavigate = jest.fn();
// const originalWindowEnv = global.window._env_;

jest.mock('../../../components/Common/LanguageSelector', () => ({
    LanguageSelector: () => <></>,
}))
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}))

window._env_ = {
    DEFAULT_FAVICON: "favicon.ico",
    DEFAULT_FONT_URL: "",
    DEFAULT_THEME: "purple_theme",
    DEFAULT_TITLE: "Inji Web Test",
    DEFAULT_LANG: 'en'
};

describe("Header Container",() => {

    test('check the presence of the three menus in the header', () => {
        render(wrapUnderRouter(<Header />));
        const headerElementLi = screen.getByTestId("Header-Menu-Elements");
        expect(headerElementLi).toBeInTheDocument();
        expect(headerElementLi.children.length).toBe(3);
    });

    // test('check the presence of the help', () => {
    //     render(wrapUnderRouter(<Header />));
    //     const headerElementLi = screen.getByTestId("Header-Menu-Help");
    //     expect(headerElementLi).toBeInTheDocument()
    //     expect(headerElementLi).toHaveTextContent("Header.help");
    //     const headerElementDiv = screen.getByTestId("Header-Menu-Help-div");
    //     expect(headerElementDiv).toBeInTheDocument()
    //
    //     jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValue(mockUseNavigate);
    //     fireEvent.click(headerElementDiv);
    //
    //     expect(mockedUsedNavigate).toHaveBeenCalled();
    //     expect(mockedUsedNavigate).toHaveBeenCalledWith("/help");
    // });
    //
    // test('check the presence of the About Inji', () => {
    //     render(wrapUnderRouter(<Header />));
    //     const headerElementLi = screen.getByTestId("Header-Menu-AboutInji");
    //     expect(headerElementLi).toBeInTheDocument()
    //     expect(headerElementLi).toHaveTextContent("Header.aboutInji");
    //     const headerElementDiv = screen.getByTestId("Header-Menu-Help-div");
    //     expect(headerElementDiv).toBeInTheDocument()
    //
    //     jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValue(mockUseNavigate);
    //     fireEvent.click(headerElementDiv);
    //
    //     expect(mockedUsedNavigate).toHaveBeenCalled();
    // });

    test('check the presence of the Langauge Selector', () => {
        render(wrapUnderRouter(<Header />));
        const headerElementLi = screen.getByTestId("Header-Menu-LanguageSelector");
        expect(headerElementLi).toBeInTheDocument()
        const headerElementDiv = screen.getByTestId("Header-Menu-Help-div");
        expect(headerElementDiv).toBeInTheDocument()

        jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValue(mockUseNavigate);
        fireEvent.click(headerElementDiv);

        expect(mockedUsedNavigate).toHaveBeenCalled();
    });
})

