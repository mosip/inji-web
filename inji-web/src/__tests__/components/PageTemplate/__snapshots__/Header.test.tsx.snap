// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Container Layout Test Check if the layout is matching with the snapshots 1`] = `
<DocumentFragment>
  <header>
    <div
      class="fixed top-0 left-0 right-0 bg-iw-background py-7 z-10"
      data-testid="Header-Container"
    >
      <div
        class="container mx-auto flex justify-between items-center px-4"
      >
        <div
          class="flex flex-row space-x-9 justify-center items-center"
          data-testid="Header-InjiWeb-Logo-Container"
        >
          <div
            class="m-3 sm:hidden "
            role="button"
            tabindex="0"
          >
            <svg
              fill="currentColor"
              height="32"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
              />
            </svg>
          </div>
          <div
            role="button"
            tabindex="0"
          >
            <img
              alt="Inji Web Logo"
              class="h-13 w-28 scale-150 cursor-pointer "
              data-testid="Header-InjiWeb-Logo"
              src="InjiWebLogo.png"
            />
          </div>
        </div>
        <nav>
          <ul
            class="flex space-x-10 items-center font-semibold"
            data-testid="Header-Menu-Elements"
          >
            <li
              data-testid="Header-Menu-Home"
            >
              <div
                class="text-iw-title cursor-pointer hidden sm:inline-block"
                data-testid="Header-Menu-Home-div"
                role="button"
                tabindex="0"
              >
                Home
              </div>
            </li>
            <li
              data-testid="Header-Menu-Help"
            >
              <div
                class=" hidden sm:block font-semibold"
                data-testid="Header-Menu-Help-div"
              >
                <div
                  class="flex flex-row justify-center items-center"
                  data-testid="HelpDropdown-Outer-Div"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="relative inline-block ms-1"
                  >
                    <button
                      class="inline-flex items-center"
                      data-testid="HelpDropdown-Button"
                      type="button"
                    >
                      <p
                        data-testid="Selected-DropDown"
                      >
                        Help
                      </p>
                      <svg
                        height="0"
                        width="0"
                      >
                        <lineargradient
                          id="blue-gradient-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                          x1="0%"
                          x2="100%"
                          y1="0%"
                          y2="100%"
                        >
                          <stop
                            offset="0%"
                            stop-color="var(--iw-color-primary)"
                          />
                          <stop
                            offset="100%"
                            stop-color="var(--iw-color-secondary)"
                          />
                        </lineargradient>
                      </svg>
                      <svg
                        color="var(--iw-color-languageArrowIcon)"
                        fill="currentColor"
                        height="20"
                        stroke="currentColor"
                        stroke-width="0"
                        style="fill: url(#blue-gradient-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA);"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 16L6 10H18L12 16Z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </nav>
        <div
          class="font-semibold"
          data-testid="Header-Menu-LanguageSelector"
        >
          <div
            class="flex flex-row justify-center items-center"
            data-testid="LanguageSelector-Outer-Div"
            role="button"
            tabindex="0"
          >
            <svg
              height="0"
              width="0"
            >
              <lineargradient
                id="blue-gradient-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="100%"
              >
                <stop
                  offset="0%"
                  stop-color="var(--iw-color-primary)"
                />
                <stop
                  offset="100%"
                  stop-color="var(--iw-color-secondary)"
                />
              </lineargradient>
            </svg>
            <svg
              color="var(--iw-color-languageGlobeIcon)"
              data-testid="Language-Selector-Icon"
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              style="fill: url(#blue-gradient-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA);"
              viewBox="0 0 16 16"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8.5 1a6.5 6.5 0 1 1 0 13 6.5 6.5 0 0 1 0-13zm4.894 4a5.527 5.527 0 0 0-3.053-2.676c.444.84.765 1.74.953 2.676h2.1zm.582 2.995A5.11 5.11 0 0 0 14 7.5a5.464 5.464 0 0 0-.213-1.5h-2.342c.032.331.055.664.055 1a10.114 10.114 0 0 1-.206 2h2.493c.095-.329.158-.665.19-1.005zm-3.535 0l.006-.051A9.04 9.04 0 0 0 10.5 7a8.994 8.994 0 0 0-.076-1H6.576A8.82 8.82 0 0 0 6.5 7a8.98 8.98 0 0 0 .233 2h3.534c.077-.332.135-.667.174-1.005zM10.249 5a8.974 8.974 0 0 0-1.255-2.97C8.83 2.016 8.666 2 8.5 2a3.62 3.62 0 0 0-.312.015l-.182.015L8 2.04A8.97 8.97 0 0 0 6.751 5h3.498zM5.706 5a9.959 9.959 0 0 1 .966-2.681A5.527 5.527 0 0 0 3.606 5h2.1zM3.213 6A5.48 5.48 0 0 0 3 7.5 5.48 5.48 0 0 0 3.213 9h2.493A10.016 10.016 0 0 1 5.5 7c0-.336.023-.669.055-1H3.213zm2.754 4h-2.36a5.515 5.515 0 0 0 3.819 2.893A10.023 10.023 0 0 1 5.967 10zM8.5 12.644A8.942 8.942 0 0 0 9.978 10H7.022A8.943 8.943 0 0 0 8.5 12.644zM11.033 10a10.024 10.024 0 0 1-1.459 2.893A5.517 5.517 0 0 0 13.393 10h-2.36z"
                fill-rule="evenodd"
              />
            </svg>
            <div
              class="relative inline-block ms-1"
            >
              <button
                class="inline-flex items-center"
                data-testid="Language-Selector-Button"
                type="button"
              >
                <p
                  data-testid="Language-Selector-Selected-DropDown-en"
                >
                  English
                </p>
                <svg
                  height="0"
                  width="0"
                >
                  <lineargradient
                    id="blue-gradient-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                    x1="0%"
                    x2="100%"
                    y1="0%"
                    y2="100%"
                  >
                    <stop
                      offset="0%"
                      stop-color="var(--iw-color-primary)"
                    />
                    <stop
                      offset="100%"
                      stop-color="var(--iw-color-secondary)"
                    />
                  </lineargradient>
                </svg>
                <svg
                  color="var(--iw-color-languageArrowIcon)"
                  fill="currentColor"
                  height="20"
                  stroke="currentColor"
                  stroke-width="0"
                  style="fill: url(#blue-gradient-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA);"
                  viewBox="0 0 24 24"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 16L6 10H18L12 16Z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
